core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -295,15 +295,16 @@ public static void writeShortDecimal(BlockBuilder blockBuilder, long value)
     public static long rescale(long value, int fromScale, int toScale)
     {
         if (toScale < fromScale) {
-            throw new IllegalArgumentException("target scale must be larger than source scale");
+            // Use MAX_SHORT_PRECISION to fit all precision long values
+            return DecimalConversions.shortToShortCast(value, MAX_SHORT_PRECISION, fromScale, MAX_SHORT_PRECISION, toScale, longTenToNth(fromScale - toScale), longTenToNth(fromScale - toScale - 1) * 5);
```

Comment:
Can we have a static import here ? 

It would be better if we could move each argument to a new line.

Reply:
Thanks for the review. Static imports done. I have split the arguments over multiple lines by grouping arguments. Let me know if one argument each line seems better.